docs/literate/src/files/adding_new_equations_literate/cubic_conservation_law_literate.jl

```diff
@@ -0,0 +1,235 @@
+#src # Adding a new equation: scalar conservation laws
+using Test #src
+
+# If you want to use Trixi for your own research, you might be interested in
+# a new physics model that's not already included in Trixi.jl. In this tutorial,
+# we will implement the cubic conservation law
+# ```math
+# \partial_t u(t,x) + \partial_x u(t,x)^3 = 0
+# ```
+# in a periodic domain in one space dimension. In Trixi.jl, such a mathematical model
+# is encoded as a subtype of [`AbstractEquations`](@trixi-ref).
+
+# ## Basic setup
+  
+# Let's start by creating a module (in the REPL, in a file, in a Jupyter notebook, ...).
+# That ensures that we can re-create `struct`s defined therein without having to
+# restart Julia.
+module CubicConservationLaw
+  
+using Trixi
+
+struct CubicEquation <: Trixi.AbstractEquations{1 #= number of spatial dimensions =#,
+                                                1 #= number of primary variables, i.e. scalar =#};
+end
+  
+end; # module
+  
+# We create `CubicEquation` as an empty `struct` since we do not use any parameters
+# for this equation. Other models could bundle arbitrary parameters, e.g., the
+# ideal gas constant for the compressible Euler equations.
+
+# From here on, the following code snippets normally are written inside the `module`.
+# To make this work outside `CubicConservationLaw.CubicEquation` is used instead of `CubicEquation`.
+# The complete code can be found at the end.
```

Comment:
We should probably rewrite these tutorials, see #851

Reply:
Since my code needs to be executable, I added the second line here and changed the code as described.
Now I rewrote it to `To make it work outside the module we use CubicConservationLaw.CubicEquation instead of CubicEquation.`, but I don't really know if this helps.
In my opinion, this is understandable if you are reading the text between the code, especially with the complete code in the right order at the end. @ranocha

Reply:
Yeah, let's revisit a possible rewriting of the tutorials later (in another PR).